ocaml/xapi/sparse_encoding.ml

```diff
@@ -67,15 +69,14 @@ module Marshal = struct
     List.iteri (fun i v ->
       Bytes.set result i (char_of_int @@ Int32.to_int v)
     ) [a; b; c; d];
-    Bytes.unsafe_to_string result
-
+    result
 end
 
 module Chunk = struct
   (** Represents an single block of data to write *)
   type t = {
     start: int64;
-    data: string;
+    data: bytes;
```

Comment:
This was effectively mutable previously due to `unmarshal` writing to it, so should be fine.